packages/core/src/build/schema.ts

```diff
@@ -28,202 +31,277 @@ export const buildSchema = ({ schema }: { schema: Schema }) => {
   });
 
   // Validate tables
-  Object.entries(getTables(schema)).forEach(([tableName, table]) => {
-    validateTableOrColumnName(tableName, "Table");
-
-    // Validate the id column
-    if (table.id === undefined)
-      throw new Error(
-        `Validation failed: Table '${tableName}' does not have an 'id' column.`,
-      );
-
-    if (isEnumColumn(table.id))
-      throw new Error(
-        `Validation failed: Invalid type for ID column '${tableName}.id'. Got 'enum', expected one of ['string', 'hex', 'bigint', 'int'].`,
-      );
-    if (isOneColumn(table.id))
-      throw new Error(
-        `Validation failed: Invalid type for ID column '${tableName}.id'. Got 'one', expected one of ['string', 'hex', 'bigint', 'int'].`,
-      );
-    if (isManyColumn(table.id))
-      throw new Error(
-        `Validation failed: Invalid type for ID column '${tableName}.id'. Got 'many', expected one of ['string', 'hex', 'bigint', 'int'].`,
-      );
-    if (isReferenceColumn(table.id))
-      throw new Error(
-        `Validation failed: Invalid type for ID column '${tableName}.id'. ID columns cannot use the '.references' modifier.`,
-      );
-
-    if (
-      table.id[" scalar"] !== "bigint" &&
-      table.id[" scalar"] !== "string" &&
-      table.id[" scalar"] !== "hex" &&
-      table.id[" scalar"] !== "int"
-    )
-      throw new Error(
-        `Validation failed: Invalid type for ID column '${tableName}.id'. Got '${table.id[" scalar"]}', expected one of ['string', 'hex', 'bigint', 'int'].`,
-      );
-
-    if (isOptionalColumn(table.id))
-      throw new Error(
-        `Validation failed: Invalid type for ID column '${tableName}.id'. ID columns cannot be optional.`,
-      );
-    if (isListColumn(table.id))
-      throw new Error(
-        `Validation failed: Invalid type for ID column '${tableName}.id'. ID columns cannot be a list.`,
-      );
-
-    // Validate all other columns
-    Object.entries(table).forEach(([columnName, column]) => {
-      if (columnName === "id") return;
-
-      validateTableOrColumnName(columnName, "Column");
-
-      if (isOneColumn(column)) {
-        const usedColumn = Object.entries(table).find(
-          ([c]) => c === column[" reference"],
+  Object.entries(getTables(schema)).forEach(
```

Comment:
Github did a bad job of rendering the diff here. I went back through and manually validated the changes